project-icp/bee-pollinator-app
# Create survey endpoints

## Overview

Create 2 endpoints to create and retrieve a Survey alongside their Subsurvey (AprilSurvey, MonthlySurvey, NovemberSurvey). These endpoints will be crucial for the Survey section of the Beekeepers app where the user will access their existing surveys as well as submit forms to save new survey data.

Connects #339, #355 

### Demo

<img width="435" alt="screen shot 2018-12-20 at 2 34 45 pm" src="https://user-images.githubusercontent.com/10568752/50307634-fa0f6900-0466-11e9-9edc-4efa77f0c51f.png">
<img width="395" alt="screen shot 2018-12-20 at 2 35 25 pm" src="https://user-images.githubusercontent.com/10568752/50307635-fa0f6900-0466-11e9-8bae-1a8110575baf.png">
<img width="485" alt="screen shot 2018-12-20 at 2 36 04 pm" src="https://user-images.githubusercontent.com/10568752/50307636-fa0f6900-0466-11e9-9516-d9a61e285e2a.png">


### Notes

I opted for view functions instead of using a DRF viewset with mixins for flexibility, though I think i could just as well have used some mixins. I don't think there's much of a difference, correct me if I'm wrong.

## Testing Instructions
- Apply migrations `./scripts/manage.sh migrate`
-  Ensure your user has at least 1 apiary, and get the `id` of one that we will create surveys on
- Visit http://localhost:8000/beekeepers/apiary/<apiary_id_here>
- Submit the mock data below, where in place of `1` in the `apiary` fields below you use your apiary's id
- You should see successful returns of the survey & subsurvey data or an error
```
{  
   "survey":{  
      "month_year":"042019",
      "num_colonies":3,
      "survey_type":"APRIL",
      "apiary":1
   },
   "colony_loss_reason":"varroa mites"
}
```

```
{  
   "survey":{  
      "month_year":"112020",
      "num_colonies":3,
      "survey_type":"NOVEMBER",
      "apiary":1
   },
   "harvested_honey":"DID_NOT_HARVEST",
   "small_hive_beetles":"False",
   "varroa_check_frequency":"NEVER",
   "varroa_manage_frequency":"NEVER"
}
```
